### PR TITLE
fix all the broken types in the r2 workers-api-reference page

### PR DESCRIPTION
**Pull request by dario-piotrowicz**
### Summary

All the types in the r2 workers-api-reference page are pretty broken and this PR is addressing them.

I am pretty sure they got broken by the starlight migration.

Thanks @aslemammad for raising this 🙂

### CODE REVIEW

File: src/content/docs/r2/api/workers/workers-api-reference.mdx [hunk 1]
```diff
@@ -304,27 +303,27 @@ Only a single hashing algorithm can be specified at once.
 
 :::
 
-* <code>md5</code>ArrayBuffer |stringoptional
+* `md5` <Type text="ArrayBuffer |string" /> <MetaInfo text="optional" />
```
Comment (KianNH) [inline src/content/docs/r2/api/workers/workers-api-reference.mdx]:
```suggestion
* `md5` <Type text="ArrayBuffer | string" /> <MetaInfo text="optional" />
```

File: src/content/docs/r2/api/workers/workers-api-reference.mdx [hunk 2]
```diff
@@ -304,27 +303,27 @@ Only a single hashing algorithm can be specified at once.
...
 
   * A md5 hash to use to check the received object's integrity.
 
-* <code>sha1</code>ArrayBuffer |stringoptional
+* `sha1` <Type text="ArrayBuffer |string" /> <MetaInfo text="optional" />
```
Comment (KianNH) [inline src/content/docs/r2/api/workers/workers-api-reference.mdx]:
```suggestion
* `sha1` <Type text="ArrayBuffer | string" /> <MetaInfo text="optional" />
```

File: src/content/docs/r2/api/workers/workers-api-reference.mdx [hunk 3]
```diff
@@ -304,27 +303,27 @@ Only a single hashing algorithm can be specified at once.
...
 
   * A SHA-1 hash to use to check the received object's integrity.
 
-* <code>sha256</code>ArrayBuffer |stringoptional
+* `sha256` <Type text="ArrayBuffer |string" /> <MetaInfo text="optional" />
```
Comment (KianNH) [inline src/content/docs/r2/api/workers/workers-api-reference.mdx]:
```suggestion
* `sha256` <Type text="ArrayBuffer | string" /> <MetaInfo text="optional" />
```

File: src/content/docs/r2/api/workers/workers-api-reference.mdx [hunk 4]
```diff
@@ -304,27 +303,27 @@ Only a single hashing algorithm can be specified at once.
...
 
   * A SHA-256 hash to use to check the received object's integrity.
 
-* <code>sha384</code>ArrayBuffer |stringoptional
+* `sha384` <Type text="ArrayBuffer |string" /> <MetaInfo text="optional" />
```
Comment (KianNH) [inline src/content/docs/r2/api/workers/workers-api-reference.mdx]:
```suggestion
* `sha384` <Type text="ArrayBuffer | string" /> <MetaInfo text="optional" />
```

File: src/content/docs/r2/api/workers/workers-api-reference.mdx [hunk 5]
```diff
@@ -304,27 +303,27 @@ Only a single hashing algorithm can be specified at once.
...
 
   * A SHA-384 hash to use to check the received object's integrity.
 
-* <code>sha512</code>ArrayBuffer |stringoptional
+* `sha512` <Type text="ArrayBuffer |string" /> <MetaInfo text="optional" />
```
Comment (KianNH) [inline src/content/docs/r2/api/workers/workers-api-reference.mdx]:
```suggestion
* `sha512` <Type text="ArrayBuffer | string" /> <MetaInfo text="optional" />
```

File: src/content/docs/r2/api/workers/workers-api-reference.mdx [hunk 6]
```diff
@@ -284,15 +283,15 @@ There are 3 variations of arguments that can be used in a range:
...
 
   * Specifies that the object should only be stored given satisfaction of certain conditions in the `R2Conditional`. Refer to [Conditional operations](#conditional-operations).
 
-* <code>httpMetadata</code>R2HTTPMetadata |Headersoptional
+* `httpMetadata` <Type text="R2HTTPMetadata |Headers" /> <MetaInfo text="optional" />
 
   * Various HTTP headers associated with the object. Refer to [HTTP Metadata](#http-metadata).
 
-* <code>customMetadata</code>Record\<string, string>optional
+* `customMetadata` <Type text="Record\<string, string>" /> <MetaInfo text="optional" />
```
Comment (KianNH) [inline src/content/docs/r2/api/workers/workers-api-reference.mdx]:
```suggestion
* `customMetadata` <Type text="Record<string, string>" /> <MetaInfo text="optional" />
```

File: src/content/docs/r2/api/workers/workers-api-reference.mdx [hunk 7]
```diff
@@ -284,15 +283,15 @@ There are 3 variations of arguments that can be used in a range:
 
 
 
-* <code>onlyIf</code>R2Conditional |Headers
+* `onlyIf` <Type text="R2Conditional | Headers" />
 
   * Specifies that the object should only be stored given satisfaction of certain conditions in the `R2Conditional`. Refer to [Conditional operations](#conditional-operations).
 
-* <code>httpMetadata</code>R2HTTPMetadata |Headersoptional
+* `httpMetadata` <Type text="R2HTTPMetadata |Headers" /> <MetaInfo text="optional" />
```
Comment (KianNH) [inline src/content/docs/r2/api/workers/workers-api-reference.mdx]:
```suggestion
* `httpMetadata` <Type text="R2HTTPMetadata | Headers" /> <MetaInfo text="optional" />
```

**Review by KianNH — APPROVED** [review]

**Comment by Aslemammad** [comment]
Thank you so much for being fast 🤝

**Pull request merged**